service/worker/migration/activities.go

```diff
@@ -681,6 +680,10 @@ func (a *activities) VerifyReplicationTasks(ctx context.Context, request *verify
 	//  - more than NonRetryableTimeout, it means potentially we encountered #4. The activity returns
 	//    non-retryable error and force-replication workflow will restarted.
 	for {
+
+		// Since replication has a lag, sleep first.
+		time.Sleep(request.VerifyInterval)
```

Comment:
this doesn't improve performance a lot but avoid 1st Verification attempts (which will most likely fail).